tools/build_api.py

```diff
@@ -547,19 +548,24 @@ def build_project(src_paths, build_path, target, toolchain_name,
         memap_instance = getattr(toolchain, 'memap_instance', None)
         memap_table = ''
         if memap_instance:
-            # Write output to stdout in text (pretty table) format
-            memap_table = memap_instance.generate_output('table', stats_depth)
-
+            real_stats_depth = stats_depth if stats_depth is None else 2
+            memap_table = memap_instance.generate_output('table', real_stats_depth)
```

Comment:
Could you hove these lines into the else statement below? that way we only generate what we need.

Reply:
memap_table is also used by the build report below, and the generate_output has side-effects I'm not sure about. You can see it was originally outside of the `if not silent` statement.

Reply:
> memap_table is also used by the build report below

That's a good point.

> generate_output has side-effects I'm not sure about

It does a `compute_report` followed by a `reduce_depth`, which will be run by the following `generate_output` calls anyway.